pkg/kfapp/ossm/types.go

```diff
@@ -28,23 +24,59 @@ type manifest struct {
 	processed bool
 }
 
-func (m *manifest) targetPath() string {
-	return fmt.Sprintf("%s%s", m.path[:len(m.path)-len(filepath.Ext(m.path))], ".yaml")
+// In order to process the templates, we need to create a tmp directory
+// to store the files. This is because embedded files are read only.
+var outputDir = "/tmp/ossm-installer/"
+
+func ensureDirExists(dir string) error {
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		err = os.MkdirAll(dir, 0755)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (m *manifest) targetPath(kfdefName string, kfdefNs string) (string, error) {
+	fullDir := filepath.Join(outputDir, kfdefNs, kfdefName, filepath.Dir(m.path))
+	if err := ensureDirExists(fullDir); err != nil {
+		return "", err
+	}
+
+	fileName := filepath.Base(m.path)
+	fileNameWithoutExt := fileName[:len(fileName)-len(filepath.Ext(fileName))]
+	return filepath.Join(fullDir, fileNameWithoutExt+".yaml"), nil
 }
 
-func (m *manifest) processTemplate(data interface{}) error {
+func (m *manifest) processTemplate(manifestRepo fs.FS, data interface{}, kfdefName string, kfdefNs string) error {
 	if !m.template {
 		return nil
 	}
-	f, err := os.Create(m.targetPath())
+	// Create file in the regular filesystem, not the embedded one
+	path, err := m.targetPath(kfdefName, kfdefNs)
 	if err != nil {
+		log.Error(err, "Failed to generate target path")
+		return err
+	}
+	f, err := os.Create(path)
+	if err != nil {
+		log.Error(err, "Failed to create file")
 		return err
 	}
 
 	tmpl := template.New(m.name).
 		Funcs(template.FuncMap{"ReplaceChar": ReplaceChar})
 
-	tmpl, err = tmpl.ParseFiles(m.path)
+	// Read file from the embedded filesystem
+	fileData, err := fs.ReadFile(manifestRepo, m.path)
```

Comment:
As commented above, I think creating writeable copy should be done before we are processing templates. Now we have two responsibilities in the function called `processTemplate`. Create tmp files and process templates. Lets move former outside.